ambry-api/src/main/java/com.github.ambry/account/AccountBuilder.java

```diff
@@ -0,0 +1,216 @@
+   * Sets the status of the {@link Account} to build.
+   * @param status The id to set.
+   * @return This builder.
+   */
+  public AccountBuilder setStatus(AccountStatus status) {
+    this.status = status;
+    return this;
+  }
+
+  /**
+   * Sets a {@link Container} for the {@link Account} to build. If the builder already has a {@link Container} with
+   * the same id as new {@link Container} to set, the new {@link Container} will replace the existing {@link Container}.
+   * @param container The new {@link Container} to set.
+   * @return This builder.
+   */
+  public AccountBuilder setContainer(Container container) {
+    if (container != null) {
+      idToContainerMetadataMap.put(container.getId(), container);
+    }
+    return this;
+  }
+
+  /**
+   * Removes a {@link Container} in this builder, so that an {@link Account} to build will not have this
+   * {@link Container}. It will be a no-op if no {@link Container} with the id exists in this builder.
+   * @param container The {@link Container} to remove.
+   * @return This builder.
+   */
+  public AccountBuilder removeContainer(Container container) {
+    if (container != null) {
+      short id = container.getId();
+      Container containerToRemove = idToContainerMetadataMap.get(id);
+      if (container.equals(containerToRemove)) {
+        idToContainerMetadataMap.remove(id);
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Sets a list of {@link Container}s for the {@link Account} to build. If any {@link Container} in the list already
+   * exists in the builder, it will be replaced by the one to set.
+   * @param containers A list of {@link Container}s to set.
+   * @return This builder.
+   */
+  public AccountBuilder setContainers(List<Container> containers) {
+    if (containers != null) {
+      for (Container container : containers) {
+        setContainer(container);
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Builds an {@link Account} object. {@code id}, {@code name}, and {@code status} are required before building.
+   * @return An {@link Account} object.
+   * @throws IllegalStateException If any required fields is not set.
+   */
+  public Account build() throws JSONException {
+    checkAccountIdInContainers();
+    checkRequiredFields(version);
+    switch (version) {
+      case ACCOUNT_METADATA_VERSION_1:
+        if (id == null) {
+          throw new IllegalStateException("Cannot build account, id is not set");
+        }
+        if (name == null) {
+          throw new IllegalStateException("Cannot build account, name is not set");
+        }
+        if (status == null) {
+          throw new IllegalStateException("Cannot build account, status is not set");
+        }
+        JSONObject jsonMetadata = new JSONObject();
+        jsonMetadata.put(ACCOUNT_METADATA_VERSION_KEY, version);
+        jsonMetadata.put(ACCOUNT_ID_KEY, id);
+        jsonMetadata.put(ACCOUNT_NAME_KEY, name);
+        jsonMetadata.put(ACCOUNT_STATUS_KEY, status);
```

Comment:
Since JSON isn't super size efficient to begin with, I think it's better to use the string representation. This means that we can more easily modify the AccountStatus enum in the future without requiring that the position of the existing enum values stay the same